### PR TITLE
Add sea trade policy on lighthouse

### PR DESCRIPTION
**Pull request by dvincent56**
- Adding Sea Trade Policy available from lighthouse
- Adding Trade policy available from advisor window
- Revamp trade price window to display policy impact on price
- Manage trade policy in city/trade_policy.c instead of caravanserai.c

sea_trade_policy

sea_trade_policy_advisor

trade_policy_price

### CODE REVIEW

**Comment by crudelios** [comment]
Some suggestions without looking at the code:

You have a duplicate "It will cost 500 Dn". If the `options_popup` has a price set as a parameter, it will automatically add that text.

On the trade prices, as long as one policy is selected, you might as well immediately show the land policy and sea policy (and not any default policy because I don't think it's needed).

You may also consider getting the `Buy price` and `Sell price` to their own line and center the text. I think right now it would make the information look cleaner and easier to understand.

**Comment by dvincent56** [comment]
#You have a duplicate "It will cost 500 Dn". If the options_popup has a price set as a parameter, it will automatically add that text.
I saw that on screenshot and I updated just before your comment ;)

For the trade price window, can you provide me an image of what you want ?

FYI, if both trade policy are the same, I just display 1 line for 2 policies
prices_same

Thank you for your feedback

**Comment by dvincent56** [comment]
Updated prices window

All status visible here
1 - no policy
2 - same policy
3 - sea policy only
4 - two different policy

all_prices_status

**Comment by crudelios** [comment]
That's starting to look better!!

Can you make the no policy/ same policy look the way they did before? (With the buy/sale price on the side and no icons)

**Comment by dvincent56** [comment]
Arf, I shifted the resources icons to center for buy/sell label alignment
If the icons and label position are different, it could be strange no ?

We can remove ship and chariot icons in this case instead ?

**Comment by crudelios** [comment]
Sorry, I know I'm being a pain... But it will look great and clean when it's done!

Yes you can remove the icons as they're unneeded in that case, but I think the centered text only looks good if there are separate land/sea prices, otherwise it does look better like the original.

Come on, you did the hard work already, it's just a bunch of `ifs` and `y` positioning now :wink:

**Comment by dvincent56** [comment]
Ok done

**Comment by crudelios** [comment]
Almost perfect, you're just missing the tooltips for the land/sea policy buttons on the advisor screen. Don't forget that you can only display the tooltips if the buttons are visible!

File: CMakeLists.txt [hunk 1]
```diff
@@ -667,7 +667,7 @@ set(SOURCE_FILES
     ${PMURHASH_FILES}
     ${PROJECT_SOURCE_DIR}/res/augustus.rc
     ${MACOSX_FILES}
-)
+        src/city/trade_policy.c src/city/trade_policy.h)
```
Comment (crudelios) [inline CMakeLists.txt:670]: Put the `)` on the next line, please.

File: src/window/trade_prices.c [hunk 2]
```diff
@@ -14,38 +15,106 @@
...
+    int sea_policy = building_monument_module_type(BUILDING_LIGHTHOUSE);
+
+    int has_land_trade_policy = has_caravanserai && land_policy && land_policy != TRADE_POLICY_3;
+    int has_sea_trade_policy = has_lighthouse && sea_policy && sea_policy != TRADE_POLICY_3;
+    int same_policy = land_policy == sea_policy;
+
+    int window_height = 11;
+    int line_buy_position = 230;
+    int line_sell_position = 270;
+    int numberMargin = 25;
```
Comment (crudelios) [inline src/window/trade_prices.c:37]: Don't use camel case. Set the variable to `number_margin`

File: src/window/trade_prices.c [hunk 3]
```diff
@@ -14,38 +15,106 @@
...
+    int button_position = 295;
+    int icon_shift = 126;
+    int price_shift = 120;
+    int fourLine = 0;
```
Comment (crudelios) [inline src/window/trade_prices.c:41]: Same here: `four_lines`

**Review by crudelios — COMMENTED** [review]
Some little code changes.

File: src/window/trade_prices.c (hunk 2 above)
Comment (dvincent56) [reply src/window/trade_prices.c:37]: professional deformation

File: CMakeLists.txt (hunk 1 above)
Comment (dvincent56) [reply CMakeLists.txt:670]: Oups should be in CITY_FILES not here

**Comment by dvincent56** [comment]
Updated

**Review by crudelios — APPROVED** [review]

**Pull request merged**